### PR TITLE
Many things

### PR DESCRIPTION
**Pull request by Chronial**

### CODE REVIEW

File: rt_dashboard/web.py [hunk 1]
```diff
@@ -256,8 +259,14 @@ def list_jobs(queue_name, page):
...
 
-        offset = (current_page - 1) * per_page
-        jobs = [serialize_job(job) for job in queue.get_tasks(offset, per_page)]
+        if reverse_order:
+            start = -1 - (current_page - 1) * per_page
+            end = start - per_page
+            jobs = reversed(queue.get_tasks(end, start))
+        else:
+            offset = (current_page - 1) * per_page
+            jobs = queue.get_tasks(offset, per_page)
```
Comment (vasilty) [inline rt_dashboard/web.py:268]: I'm not sure I understand why `per_page` is the second argument. If `current_page` is 2, then we call `queue.get_tasks(per_page, per_page)`, is that correct?

Comment (vasilty) [reply rt_dashboard/web.py:268]: Or wait, it is the `length` argument, which explains it. Let me think again about the reverse order :)

File: rt_dashboard/templates/rt_dashboard/dashboard.html [hunk 2]
```diff
@@ -103,7 +103,14 @@ <h1>Tasks on <strong{% if queue.name == '[failed]' %} class="failed"{% endif %}>
                 <i class="icon-trash icon-white"></i> Delete
             </a>
         {% endif %}
-        This list below contains all the registered tasks on queue <strong>{{ queue.name }}</strong>, sorted by age (oldest on top).</p>
+        This list below contains all the registered tasks on queue
+        <strong>{{ queue.name }}</strong>, sorted by age
+        {% if queue.name.startswith('[') %}
+            (newest on top).
```
Comment (vasilty) [inline rt_dashboard/templates/rt_dashboard/dashboard.html:109]: This won't be necessarily right for the `[running]` queue, right?

File: rt_dashboard/history.py [hunk 3]
```diff
@@ -25,12 +24,21 @@ def task_tooltip(t):
 
 
 def get_history_context():
-    tz = pytz.timezone(settings.TIMEZONE)
-    tasks = finished_task_registry.get_tasks() + failed_task_registry.get_tasks()
-    start = utcnow() - datetime.timedelta(days=2)
-    tasks = [t for t in tasks if t.started_at > start]
+    max_finished_tasks = 5000
```
Comment (vasilty) [inline rt_dashboard/history.py:31]: @i-trofimtschuk Would this work for yf, as yf seems to have more tasks per day than that?

**Review by vasilty — COMMENTED** [review]
Looks nice! I'm looking forward to those changes. I had to spend some time thinking about the `queue.get_tasks()`, somehow the arguments and values passed in didn't make sense in the beginning, but I believe I got it now. I also tested the changes a bit: displaying of the currently running tasks in the history view as well as ordering by the newest tasks first in the failed/finished queue are absolutely amazing.

File: rt_dashboard/history.py (hunk 3 above)
Comment (i-trofimtschuk) [reply rt_dashboard/history.py:31]: we have ~90k-100k per week so more like 15000

File: rt_dashboard/templates/rt_dashboard/dashboard.html (hunk 2 above)
Comment (Chronial) [reply rt_dashboard/templates/rt_dashboard/dashboard.html:109]: That is correct, but I chose to ignore that for now. It was also wrong before.

File: rt_dashboard/history.py (hunk 3 above)
Comment (Chronial) [reply rt_dashboard/history.py:31]: I had the impression that ston is close to the limit of what the system can handle, that's why I chose this number. I'm pretty sure that yf just has too many tasks per day to display them all. But let's deploy this for now and then we can see whether there is room for displaying more tasks.

Comment (vasilty) [reply rt_dashboard/history.py:31]: I agree with that. In any case 5000 tasks being displayed is better than no tasks as is currently the case at yf

**Pull request merged**